### PR TITLE
Convert `datetime.date` to int64 (nanosec from epoch)

### PR DESCRIPTION
**Pull request by selitvin**
`pyarrow==0.12.1` now reads parquet `DateType` columns as python datetime object.
TF does not support conversion of this native python time. We convert datetime.date column
to the number of nano-seconds since epoch in int64 to make the column into TF friendly.

### CODE REVIEW

File: petastorm/tf_utils.py [hunk 1]
```diff
@@ -69,13 +71,20 @@ def _sanitize_field_tf_types(sample):
...
             next_sample_dict[k] = v.astype(np.int32)
         elif isinstance(v, np.ndarray) and v.dtype.type in (np.bytes_, np.unicode_):
             if v.size != 0:
                 next_sample_dict[k] = v.tolist()
+        elif isinstance(v, np.ndarray) and v.dtype.kind == 'O' and isinstance(v[0], datetime.date):
+            # Pyarrow 0.12.1 started returning python datetime.date when parquet column is a DateType() column.
+            # Convert values in such column into nsec from epoch int64.
+            def date_to_nsec_from_epoch(dt):
+                return timegm(dt.timetuple()) * 1000000000
+            to_nsec = np.vectorize(date_to_nsec_from_epoch)
```
Comment (SashaOv) [inline petastorm/tf_utils.py]: This is executed for every value. Since this is constant, it would make sense to define it outside of this function

File: petastorm/tests/test_tf_utils.py [hunk 2]
```diff
@@ -46,13 +48,17 @@ def _tf_session():
         coord.join(threads)
 
 
-def test_empty_dict():
-    # Check two types that should be promoted/converted (uint16 and Decimal) and one that should not be
-    # modified (int32)
+def test_sanitize_field_tf_types():
+    expected_datetime_array = [datetime.date(1970, 1, 1), datetime.date(2015, 9, 29)]
+    expected_datetime_ns_from_epoch = [timegm(dt.timetuple()) * 1000000000 for dt in expected_datetime_array]
```
Comment (SashaOv) [inline petastorm/tests/test_tf_utils.py]: Wanna add a test with non-zero nanoseconds?

Comment (selitvin) [reply petastorm/tests/test_tf_utils.py]: I don't think datetime gets to the nanoseconds resolution...
